### ISSUE TITLE
HTTP/2 GOAWAY logic doesn't support graceful shutdown

### ISSUE THREAD

**Issue by nmittler** [netty/netty#3564]
From [section 6.8](https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.8):

> A server that is attempting to gracefully shut down a
   connection SHOULD send an initial GOAWAY frame with the last stream
   identifier set to 2^31-1 and a NO_ERROR code.  This signals to the
   client that a shutdown is imminent and that no further requests can
   be initiated.  After waiting at least one round trip time, the server
   can send another GOAWAY frame with an updated last stream identifier.
   This ensures that a connection can be cleanly shut down without
   losing requests.

We need to support sending 2 GOAWAY frames in the case of graceful shutdown.

**Comment by Scottmitch** [comment]
@buchgr's PR https://github.com/netty/netty/pull/3516 will handle this I think.

**Comment by buchgr** [comment]
@Scottmitch yes. It will allow sending multiple GOAWAY frames as long as the last stream identifier is less than the last stream identifier of the previous GOAWAY frame. So this should work.

**Comment by Scottmitch** [comment]
Closing PR https://github.com/netty/netty/pull/3516 has been cherry-picked.

**Comment by rockeshub** [comment]
Hi @buchgr @nmittler @Scottmitch, 
I know the issue is closed, But have one doubt regarding [section 6.8](https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.8):

Is it mandatory server (here netty) to send multiple GOAWAYs always ?

```
 A client that is unable to retry requests loses all requests that are
   in flight when the server closes the connection.  This is especially
   true for intermediaries that might not be serving clients using
   HTTP/2. A server that is attempting to gracefully shut down a
connection SHOULD send an initial GOAWAY frame with the last stream
identifier set to 2^31-1 and a NO_ERROR code. This signals to the
client that a shutdown is imminent and that no further requests can
be initiated. After waiting at least one round trip time, the server
can send another GOAWAY frame with an updated last stream identifier.
This ensures that a connection can be cleanly shut down without
losing requests.
```
From the above statement, it says two GOAWAYs should be sent.
I am asking this as of today when Netty is gracefully shutting down. it sends the first GOAWAY  with lastStreamId as the lastStreamId known to the server and then waits for the request to process and then closes the connection.
is this the right behavior?

Below is the pcap that I captured when sending a request to the spring cloud gateway which is using the Netty server. and request delay is 8 seconds
![image](https://github.com/netty/netty/assets/17350044/78736f63-45b9-4af1-8d22-1f97cbbea994)
